tofhir-engine/src/test/scala/io/tofhir/test/engine/execution/FileStreamInputArchiverTest.scala

```diff
@@ -0,0 +1,129 @@
+package io.tofhir.test.engine.execution
+
+import io.tofhir.engine.config.ToFhirConfig
+import io.tofhir.engine.execution.{FileStreamInputArchiver, RunningJobRegistry}
+import io.tofhir.engine.model._
+import io.tofhir.engine.util.FileUtils
+import org.mockito.MockitoSugar._
+import org.scalatest.flatspec.AnyFlatSpec
+import org.scalatest.matchers.should.Matchers
+
+import java.io.PrintWriter
+
+class FileStreamInputArchiverTest extends AnyFlatSpec with Matchers {
+
+  val runningJobRegistryMock = mock[RunningJobRegistry]
+  val fileStreamInputArchiver = new FileStreamInputArchiver(runningJobRegistryMock)
+
+  "FileStreamInputArchiver" should "apply archiving for a streaming job" in {
+
+    // Create a mock execution
+    val mockTaskExecution = mock[FhirMappingJobExecution]
+    val mockJob = mock[FhirMappingJob]
+    val mockDataProcessingSettings = mock[DataProcessingSettings]
+    val mappingUrl = "mocked_mapping_url"
+    val jobId = "mocked_job_id"
+    when(mockTaskExecution.id).thenReturn(jobId)
+    when(mockTaskExecution.job).thenReturn(mockJob)
+    when(mockJob.dataProcessingSettings).thenReturn(mockDataProcessingSettings)
+    when(mockDataProcessingSettings.archiveMode).thenReturn(ArchiveModes.ARCHIVE)
+    when(mockTaskExecution.getCheckpointDirectory(mappingUrl)).thenReturn(
+      FileUtils.getPath("test-archiver", jobId, mappingUrl.hashCode.toString).toString)
+
+
+    // Create a source file to refer location of test.csv
+    val sourceFile = FileUtils.getPath("test-archiver", jobId, mappingUrl.hashCode.toString, "sources", "0", "0").toFile
```

Comment:
We can extract the logic to access to source folder of Spark and add a dedicated test for that